core/templates/base-components/warning-loader.directive.ts

```diff
@@ -26,11 +27,19 @@ angular.module('oppia').directive('warningLoader', [
       bindToController: {},
       template: require('./warning-loader.directive.html'),
       controllerAs: '$ctrl',
-      controller: ['AlertsService',
-        function(AlertsService) {
+      controller: ['AlertsService', '$rootScope', 'LoaderService',
+        function(AlertsService, $rootScope, LoaderService) {
           var ctrl = this;
           ctrl.$onInit = function() {
             ctrl.AlertsService = AlertsService;
+            /**
+             * TODO(@srijanreddy98): when migrating to angular 8
```

Comment:
Note that this is not the correct syntax for TODOs. Use an issue number instead.

Also, I don't understand why this code is sitting in the "warning loader". That seems wrong, conceptually? The loading message is not a warning.

Reply:
> Note that this is not the correct syntax for TODOs. Use an issue number instead.

Should I create an issue for this?

> Also, I don't understand why this code is sitting in the "warning loader". That seems wrong, conceptually? The loading message is not a warning.
True, But this is where the loading message is shown. `warning-loader.directive.html` line number 21.

Reply:
Also in this particular case, I haven't found a pattern like the other TODOs in the changes, should I still create an issue for this?

Reply:
Oh, I see -- yes, you're right. Do you think it might be worth separating into warning-and-alerts.directive.html and loading-message.directive.html since they're pretty different? I think you can make that call.

> Also in this particular case, I haven't found a pattern like the other TODOs in the changes, should I still create an issue for this?

Sorry, I don't understand -- what do you mean by "I haven't found a pattern like the other TODOs in the changes"?

Reply:
I meant like the other places where I replaced TODO's with an issue number. Issues #9154 and #9155 were created for a large range of code changes that might happen due to migration. This seems to all alone in a league of its own.

Reply:
Ah, well -- there are really only two cases for handling TODOs.

- Either (preferably): do it in the PR itself and don't have a TODO at all.
- Or, if that's not possible, file an issue and have the TODO link to the issue. If it makes sense, assign the issue to yourself for follow-up at the appropriate time.

These are the only two cases. We've found that "TODO(author)" comments in the codebase simply don't get resolved, after years.